website/project/model.py

```diff
@@ -3621,81 +3681,97 @@ class DraftRegistration(AddonModelMixin, StoredObject):
+                for old_comment in old_comments:
+                    for new_comment in new_comments:
+                        new_id = new_comment.get('created', [])
+                        # if the primary key is already in use and the old comment is more recent,
+                        if new_id in old_comment_ids and old_comment['lastModified'] > new_comment['lastModified']:
+                            # use the old one instead
+                            loc = new_comments.index(new_comment)
+                            new_comments[loc] = old_comment
+
             if not old_value or old_value.get('value') != value.get('value'):
-                changes.append(key)
+                changes.append(question_id)
+
         self.registration_metadata.update(metadata)
+
         if save:
             self.save()
-        # TODO: uncomment to nullify approval state if edited
-        # project_signals.draft_edited.send(self, changes)
-        # self.after_edit(changes)
+        project_signals.draft_edited.send(changes)
+        self.after_edit(changes)
         return changes
 
     def before_edit(self, auth):
+        # TODO(samchrisinger): Make sure we still need this
         messages = []
-        # TODO: uncomment to provide pre-edit warnings for drafts that are approved or are pending approval
-        # if self.flags.get('isApproved'):
-        #     messages.append('The draft registration you are editing is currently approved. Please note that if you make any changes (excluding comments) this approval status will be revoked and you will need to submit for approval again.')
-        # if self.flags.get('isPendingReview'):
-        #     messages.append('The draft registration you are editing is currently pending review. Please note that if you make any changes (excluding comments) this request will be cancelled and you will need to submit for approval again.')
+        if self.is_approved:
+            messages.append('The draft registration you are editing is currently approved. Please note that if you make any changes (excluding comments) this approval status will be revoked and you will need to submit for approval again.')
+        if self.flags.get('isPendingReview'):
+            messages.append('The draft registration you are editing is currently pending review. Please note that if you make any changes (excluding comments) this request will be canceled and you will need to submit for approval again.')
         return messages
 
     def after_edit(self, changes):
-        pass
-        # TODO: uncomment when we support review/approval
-        ## revoke approval and review status if changed
-        #if changes:
-        #    self.flags.update({
-        #        'isPendingReview': False,
-        #        'isApproved': False
-        #    })
-        #    self.save()
+
+        if changes:
+            self.flags.update({
+                'isPendingReview': False,
+                'isApproved': False
```

Comment:
So there are entire class methods build around this one use case? Oomph. Will this lead to dead code after prereg is done?

(thanks for the discussions and clarifications, incidentally. Really appreciate your insights)